src/rtc.rs

```diff
@@ -158,12 +158,24 @@ impl RtcConfig {
 }
 
 impl Rtc {
+    #[allow(clippy::self_named_constructors)]
+    #[deprecated = "use `new()` instead"]
```

Comment:
This is potentially controversial. Opinions?
Also has conflicts with #297 (whichever is merged last will need modifications to suppress the deprecation warnings assuming this remains)